### PR TITLE
Add Energy panel

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

First version of the energy panel dashboard, still needs cleanup and polish, but it is already huge as it is now.

Needs https://github.com/home-assistant/core/pull/52001

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/chart/statistics-charts.ts [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    if (!isComponentLoaded(this.hass, "history")) {
+      return html`<div class="info">
+        ${this.hass.localize("ui.components.history_charts.history_disabled")}
+      </div>`;
+    }
+
+    if (this.isLoadingData && !this.statisticsData) {
+      return html`<div class="info">
+        ${this.hass.localize(
+          "ui.components.statistics_charts.loading_statistics"
```
Comment (balloob) [inline src/components/chart/statistics-charts.ts:135]: We can share these strings with the history-based chart. It's all data. The user doesn't have to know there is a difference between both.

Comment (bramkragten) [reply src/components/chart/statistics-charts.ts:135]: It is all data, but statistics can also be future and not just history ;-)

**Review by balloob — DISMISSED** [review]
super vet

**Comment by spacegaier** [comment]
Pics or it did not happen 😆.

**Comment by bramkragten** [comment]
> Pics or it did not happen 😆.

Not much to show yet, it is just the base :-)

File: package.json [hunk 2]
```diff
@@ -44,6 +44,7 @@
     "@fullcalendar/list": "5.1.0",
     "@lit-labs/virtualizer": "patch:@lit-labs/virtualizer@0.6.0#./.yarn/patches/@lit-labs/virtualizer/0.7.0.patch",
     "@material/chips": "=12.0.0-canary.1a8d06483.0",
+    "@material/data-table": "=12.0.0-canary.1a8d06483.0",
```
Comment (balloob) [inline package.json:47]: This is different than below version

File: src/data/history.ts [hunk 3]
```diff
@@ -294,6 +299,48 @@ export const fetchStatistics = (
...
+  if (startSum === null) {
+    return endSum;
+  }
+  return endSum - startSum;
+};
+
+export const calculateStatisticsSumGrowth = (
+  data: Statistics,
+  stats: string[]
+): number | undefined => {
```
Comment (balloob) [inline src/data/history.ts:325]: Maybe also make this `null` to align with the singular variant.

File: src/panels/lovelace/cards/hui-energy-solar-graph-card.ts [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+    super.connectedCallback();
+    if (!this.hasUpdated) {
+      return;
+    }
+    this._getStatistics();
+    // statistics are created every hour
+    clearInterval(this._interval);
+    this._interval = window.setInterval(
+      () => this._getStatistics(),
+      1000 * 60 * 60
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-energy-solar-graph-card.ts:78]: Statistics are created every hour at 12 past whole. In the future let's schedule an interval at 20 past or something.

File: src/panels/lovelace/views/hui-sidebar-view.ts [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+  LovelaceViewConfig,
+  LovelaceViewElement,
+} from "../../../data/lovelace";
+import type { HomeAssistant } from "../../../types";
+import { HuiErrorCard } from "../cards/hui-error-card";
+import type { Lovelace, LovelaceCard } from "../types";
+
+let editCodeLoaded = false;
+
+export class SideBarView extends LitElement implements LovelaceViewElement {
```
Comment (balloob) [inline src/panels/lovelace/views/hui-sidebar-view.ts:24]: This one will be popular. Let's make sure we document it.

**Review by balloob — DISMISSED** [review]

File: package.json (hunk 2 above)
Comment (balloob) [reply package.json:47]: ah mwc vs mdc.

**Review by balloob — APPROVED** [review]

**Pull request merged**